internal/pkg/agent/application/upgrade/step_unpack.go

```diff
@@ -153,26 +263,25 @@ func untar(log *logger.Logger, version string, archivePath, dataDir string) (str
 			break
 		}
 		if err != nil {
-			return "", err
+			return UnpackResult{}, err
 		}
 
 		if !validFileName(f.Name) {
-			return "", errors.New("tar contained invalid filename: %q", f.Name, errors.TypeFilesystem, errors.M(errors.MetaKeyPath, f.Name))
+			return UnpackResult{}, errors.New("tar contained invalid filename: %q", f.Name, errors.TypeFilesystem, errors.M(errors.MetaKeyPath, f.Name))
 		}
 
-		//get hash
 		fileName := strings.TrimPrefix(f.Name, fileNamePrefix)
 
 		if fileName == agentCommitFile {
-			hashBytes, err := io.ReadAll(tr)
-			if err != nil || len(hashBytes) < hashLen {
-				return "", err
-			}
-
-			hash = string(hashBytes[:hashLen])
 			continue
```

Comment:
Could you add a comment here that we already loaded the hash, similar to what you did in the zip code path? Thanks.

Reply:
Done in https://github.com/elastic/elastic-agent/pull/4174/commits/9f2a4cc546d99c58e4e4325a46a97abedbffb874